modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/http/routes/NodeParametersRoutes.scala

```diff
@@ -52,6 +59,56 @@ final case class NodeParametersRoutes[F[_]: Async](
       maybeSnapshot <- getLatestFullSnapshot
     } yield maybeSnapshot.flatMap(_.value.info.updateNodeParameters.flatMap(_.get(nodeId)))
 
+  @derive(eqv, show, encoder)
+  case class NodeParametersInfo(
+    node: PeerInfo,
+    delegatedStakeRewardParameters: DelegatedStakeRewardParameters,
+    nodeMetadataParameters: NodeMetadataParameters
+  )
+
+  sealed trait SortOrder
+  object SortOrder {
+    case object Asc extends SortOrder
+    case object Desc extends SortOrder
+    def fromString(s: String): Option[SortOrder] = Option(s).getOrElse("").toLowerCase match {
+      case "asc"  => Some(Asc)
+      case "desc" => Some(Desc)
+      case _      => None
+    }
+  }
+  sealed trait Sort
+  object Sort {
+    case object Address extends Sort
+    case object Name extends Sort
+    case object PeerId extends Sort
+    def fromString(s: String): Option[Sort] = Option(s).getOrElse("").toLowerCase match {
+      case "address" => Some(Address)
+      case "name"    => Some(Name)
+      case "peerid"  => Some(PeerId)
+      case _         => None
+    }
+  }
+
+  private def sort(list: List[NodeParametersInfo], sortOrder: Option[SortOrder], sort: Option[Sort]): F[List[NodeParametersInfo]] = {
+    def applyOrder(sorted: List[NodeParametersInfo], order: Option[SortOrder]): List[NodeParametersInfo] =
+      order match {
+        case Some(SortOrder.Desc) => sorted.reverse
+        case _                    => sorted
+      }
+
+    (sort match {
+      case None               => list.pure[F]
+      case Some(Sort.Address) => list.traverse(info => info.node.id.toAddress.map(addr => (info, addr))).map(_.sortBy(_._2).map(_._1))
+      case Some(Sort.Name)    => list.sortBy(_.nodeMetadataParameters.name).pure[F]
+      case Some(Sort.PeerId)  => list.sortBy(_.node.id.show).pure[F]
+    }).map(sorted => applyOrder(sorted, sortOrder))
+  }
+
+  private def filter(list: List[NodeParametersInfo], search: String): List[NodeParametersInfo] =
+    if (search == "") list
+    else
+      list.filter(info => info.nodeMetadataParameters.name.contains(search) || info.node.id.show == search)
```

Comment:
Comparison should use id.value.value to get String directly rather than derived show.